src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -795,8 +794,11 @@ private long approximateLastCycle2(int lastCycle) throws StreamCorruptedExceptio
                     // try again.
                 }
             }
+            this.setCycle(lastCycle);
             return rollCycle.toIndex(lastCycle, 0L);
```

Comment:
If we get here, lastCycle == prevCycle? would it be clearer to 
```java
this.setCycle(prevCycle);
return rollCycle.toIndex(prevCycle, 0L);
```

Also should `prevCycle` be called `firstCycle` to make it clearer what's going on here?